OpenSim/Common/Storage.cpp

```diff
@@ -2857,13 +2857,22 @@ print(const string &aFileName,double aDT,const string &aMode) const
     return(nTotal);
 }
 
-void Storage::
-printResult(const Storage *aStorage,const std::string &aName,
-                const std::string &aDir,double aDT,const std::string &aExtension)
-{
-    if(!aStorage) return;
-    std::string path = (aDir=="") ? "." : aDir;
-    std::string name = (aName.rfind(aExtension)==string::npos)? (path + "/" + aName + aExtension) :  (path + "/" + aName);
+void Storage::printResult(const Storage* aStorage, const std::string& aName,
+        const std::string& aDir, double aDT, const std::string& aExtension) {
+    if (!aStorage) return;
+    std::string path = (aDir == "") ? "." : aDir;
+    std::string directory;
+    bool dontApplySearchPath;
+    std::string fileName, extension;
+    SimTK::Pathname::deconstructPathname(
+            aName, dontApplySearchPath, directory, fileName, extension);
+    if (directory != "") {
+        log_warn("Directory {} was specified where only file name {} is expected. It will "
```

Comment:
- Suggest "It will be ignored" -> "The directory will be ignored" (i.e. not the file name).
- Perhaps quotes around all three `{}` (as was done for the third one) to offset data from text.